data-entry/invoice-editor/invoice-editor-flow/src/main/java/com/vaadin/invoice/editor/InvoiceEditor.java

```diff
@@ -142,21 +143,36 @@ public InvoiceEditor() {
 
         // Grid Pro
         GridPro<Invoice> grid = new GridPro<>();
-        grid.setItems(createItems());
+        List<Invoice> invoiceList = createItems();
+        grid.setItems(invoiceList);
+
+        addCardTransactionBtn.addClickListener(e -> {
+            invoiceList.add(0, new Invoice("", 0, Currency.EUR, 0, 0, Category.PERSONAL, false, 0));
```

Comment:
Currently, if you add a new invoice and edit some other than first field, the notification says:
"Description was updated to be: foobar for product null"
So maybe should also set the product name to empty string instead of null? (maybe it should also be a constructor parameter for Invoice)

Reply:
Updated. I have not been thinking about the app logic and the proper messages.